src/main/java/Duke.java

```diff
@@ -1,10 +1,158 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
 public class Duke {
+    static final String LOGO = "$$$$$$$\\                  $$\\           \n" +
+            "$$  __$$\\                 $$ |          \n" +
+            "$$ |  $$ |$$\\   $$\\  $$$$$$$ | $$$$$$\\  \n" +
+            "$$ |  $$ |$$ |  $$ |$$  __$$ |$$  __$$\\ \n" +
+            "$$ |  $$ |$$ |  $$ |$$ /  $$ |$$$$$$$$ |\n" +
+            "$$ |  $$ |$$ |  $$ |$$ |  $$ |$$   ____|\n" +
+            "$$$$$$$  |\\$$$$$$  |\\$$$$$$$ |\\$$$$$$$\\ \n" +
+            "\\_______/  \\______/  \\_______| \\_______|";
+    static final String GREET_MESSAGE = "I am Dude.\n" +
+            "Type something in, dude.";
+    static final String BYE_MESSAGE = "Catch you later, dude.";
+
+    static List<Task> tasks = new ArrayList<>();
+
+    public static void runCommand(String command) {
+        String[] args = command.split(" ", 2);
+        String action = args[0];
+        switch (action) {
+        case "bye":
+            System.out.println(BYE_MESSAGE);
+            System.exit(0);
+            // no break needed, the code has already exited
+        case "list":
+            listTasks();
+            break;
+        case "mark":
+        case "unmark": {
+            boolean isDone = action.equals("mark");
+            if (args.length < 2) {
+                System.out.println("Not enough arguments, dude.");
+                break;
+            }
+            int index;
+            try {
+                index = Integer.parseInt(args[1].trim()) - 1;
+            } catch (NumberFormatException e) {
+                // user either didn't enter a number or number was too big for integer
+                // not a nice way to use exceptions but alternatives are cumbersome
+                System.out.println("Invalid argument, dude.");
+                break;
+            }
+            setTaskStatus(index, isDone);
+            break;
+        }
+        case "todo":
+        case "deadline":
+        case "event": {
+            if (args.length < 2) {
+                System.out.println("Not enough arguments, dude."); // repeated code, can this be collapsed?
+                break;
+            }
+            String taskData = args[1];
+            addTask(action, taskData);
+            break;
+        }
+        default:
+            System.out.println("I don't recognise that command, dude.");
+            break;
+        }
+    }
+
+    private static void addTask(String taskType, String taskData) {
+        Task task;
+        switch (taskType) {
+        case "todo": {
+            task = createTodo(taskData);
+            break;
+        }
```

Comment:
If I were to declare a variable with the same name in two different cases, the braces are necessary to distinguish them as being as different scopes, or else Java complains about variable redefinition within the same scope. In this case I have fortuitously avoided that scenario, but this will probably come up as I add more code, and I'm not sure what the best way to handle this apart from putting braces around every case statement (using if statements instead? defining common variables in case statements just outside the switch statement?)